plugins/src/main/java/org/smartregister/fhir/gateway/plugins/SyncAccessDecision.java

```diff
@@ -357,16 +357,28 @@ private Bundle postProcessModeListEntries(
 
         Bundle resultBundle = fhirR4Client.transaction().withBundle(requestBundle).execute();
 
-        // add total
-        resultBundle.setTotal(requestBundle.getEntry().size());
+        StringBuilder urlBuilder = new StringBuilder(request.getFhirServerBase());
+        urlBuilder.append("/").append(request.getRequestPath());
 
-        // add pagination links
-        int nextPage = page < totalEntries / count ? page + 1 : 0; // 0 indicates no next page
+        return addPaginationLinks(urlBuilder, resultBundle, page, totalEntries, count, parameters);
+    }
+
+    public static Bundle addPaginationLinks(
```

Comment:
Since this method is being used by multiple classes let us create and move it to a generic Utils class.